libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -386,112 +388,130 @@ internal object MapboxNavigationTelemetry {
         @FeedbackEvent.Source feedbackSource: String,
         screenshot: String?,
         feedbackSubType: Array<@FeedbackEvent.SubType String>?,
+        feedbackMetadata: FeedbackMetadata? = null,
     ) {
         createUserFeedback(
             feedbackType,
             description,
             feedbackSource,
             screenshot,
             feedbackSubType,
-            null
+            feedbackMetadata,
+            null,
         ) {
             sendMetricEvent(it)
         }
     }
 
-    fun cacheUserFeedback(
-        @FeedbackEvent.Type feedbackType: String,
-        description: String,
-        @FeedbackEvent.Source feedbackSource: String,
-        screenshot: String?,
-        feedbackSubType: Array<String>?,
-    ) {
-        createUserFeedback(
-            feedbackType,
-            description,
-            feedbackSource,
-            screenshot,
-            feedbackSubType,
-            {
-                feedbackEventCacheMap[it.feedbackId] = it
-            },
-            {
-                feedbackEventCacheMap[it.feedbackId]?.let { cachedEvent ->
-                    cachedEvent.locationsAfter = it.locationsAfter
-                    cachedEvent.locationsBefore = it.locationsBefore
-                } ?: feedbackEventCacheMap.put(it.feedbackId, it)
-            }
-        )
-    }
-
-    fun getCachedUserFeedback(): List<CachedNavigationFeedbackEvent> {
-        locationsCollector.flushBuffers()
-        return feedbackEventCacheMap.map {
-            it.value.getCachedNavigationFeedbackEvent()
-        }
-    }
-
-    fun postCachedUserFeedback(cachedFeedbackEventList: List<CachedNavigationFeedbackEvent>) {
-        log("post cached user feedback events")
-        val feedbackEventCache = LinkedHashMap(feedbackEventCacheMap)
-        feedbackEventCacheMap.clear()
-
-        cachedFeedbackEventList.forEach { cachedFeedback ->
-            sendEvent(
-                feedbackEventCache[cachedFeedback.feedbackId]?.apply {
-                    update(cachedFeedback)
-                } ?: return@forEach
-            )
-        }
-    }
-
+    @ExperimentalPreviewMapboxNavigationAPI
     private fun createUserFeedback(
         @FeedbackEvent.Type feedbackType: String,
         description: String,
         @FeedbackEvent.Source feedbackSource: String,
         screenshot: String?,
         feedbackSubType: Array<@FeedbackEvent.SubType String>?,
-        onEventCreated: ((NavigationFeedbackEvent) -> Unit)? = null,
-        onEventUpdated: ((NavigationFeedbackEvent) -> Unit)? = null
+        feedbackMetadata: FeedbackMetadata?,
+        onEventCreated: ((NavigationFeedbackEvent) -> Unit)?,
+        onEventUpdated: ((NavigationFeedbackEvent) -> Unit)?,
     ) {
-        ifTelemetryRunning(
-            "User Feedback event creation failed: $LOG_TELEMETRY_IS_NOT_RUNNING"
-        ) {
-            log("collect post event locations for user feedback")
+        log("collect post event locations for user feedback")
```

Comment:
This log should be moved to `if (feedbackMetadata == null)` and another log should be done if the event is immediately sent with metadata.